packages/phpstan-rules/src/Rules/PreventDuplicateClassMethodRule.php

```diff
@@ -106,19 +118,20 @@ public function process(Node $node, Scope $scope): array
             return [];
         }
 
-        $printStmts = $this->printerStandard->prettyPrint($stmts);
+        $printStmts = $this->getPrintStmts($node);
+        $append     = sprintf(self::PHPSTAN_RULES_METHOD_PARAM_COUNT, count($node->params));
```

Comment:
Where is the param count needed? It seems this can be removed completely, I can't find any related test

Reply:
that needed to differentiate between same method, but different parameter. In fixture test, the method is named "someMethod", with and without parameter. Without parameter count, all will point to 1st "someMethod" found that can have different parameter to compare, and comparison will be invalid.

Reply:
In test fixture I see only `someMethod()` both with 1 parameter.
But even if the parameter count is different and the body is the same, it's still duplicate.

Reply:
the FirstClass fixture has no parameter, that method is a comparison, read first, if no count param check, all will compare with FirstClass method, so mask name check won't work without that.

Reply:
see FirstClass method for comparison

https://github.com/symplify/symplify/blob/f7fbbca9a4eb388c53e0d62467a687c263fe462e/packages/phpstan-rules/tests/Rules/PreventDuplicateClassMethodRule/Fixture/FirstClass.php#L15

Reply:
That's not very clear from tests. Could you add new test fixture with their own files, where this can be spot?

Reply:
I've added new fixture for demo differentiate between has and not has parameter for method point to compare https://github.com/symplify/symplify/pull/2744/commits/8a5eb790fc772607d286351d8bc77348470b798c

Reply:
I see better now. Thank you!


Could you add one test fixture with 2 methods? This should report 1 error:

```php
class FirstVersionWithoutParam
{
    public function go($value)
    {
        return $value + 1000;
    }
}
```

```php

class FirstVersionWithoutParam
{
    public function sleep($another, $yet)
    {
        return $yet + 1000;
    }
}
```

Reply:
that's `$yet + 1000` is a single statement as far as i know, so that will be skipped as the following check:

https://github.com/symplify/symplify/blob/f7fbbca9a4eb388c53e0d62467a687c263fe462e/packages/phpstan-rules/src/Rules/PreventDuplicateClassMethodRule.php#L103-L107

Also, the method is different ( `go` and `sleep` ), we previously only compare if method is same, do we need to check content same with different method as well?

Reply:
> that's $yet + 1000 is a single s

Ah, my bad. There should be one extra line in both cases then, e.g.

```php
$yet += 100000;
return $yet + 1000;
```

> do we need to check content same with different method as well?

Yes, we should apply same logic as we do with variables. If the logic inside method is the same, it's duplicated method.

It's like with phones. It doesn't make if it's iPhone or Samsungs, its still a phone.

Reply:
`go` and `sleep` have different count of parameters, the previous behaviour won't work

Reply:
That's allright. Go for it

Reply:
I will refactor it to allow different name with same count parameter.

Reply:
I've refactored to detect different method name content with same count parameter